### PR TITLE
KOGITO-4444 - Sanitize dashboard name

### PR DESCRIPTION
**Pull request by r00ta**
JIRA: https://issues.redhat.com/browse/KOGITO-4444

The codegen generates valid dashboard filenames (for example `Traffic Violation.json`) based on the DMN model name, but these are not valid names for the grafana operator. This is why we need to sanitize the dashboard name before we create the `GrafanaDashboard`. 

We might change the filename on codegen side as well, but since it generates valid filenames, I think it's ok for the time being. 

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [x] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

**Review by ricardozanini — APPROVED** [review]

**Comment by ricardozanini** [comment]
Do not merge until #740 gets merged

File: pkg/infrastructure/services/dashboards.go
Comment (spolti) [inline pkg/infrastructure/services/dashboards.go:160]: this func will remove any - or _ characters, is it expected to happen?

**Review by spolti — CHANGES_REQUESTED** [review]

**Comment by sutaakar** [comment]
@r00ta Can you please answer and rebase?

File: pkg/infrastructure/services/dashboards.go
Comment (r00ta) [reply pkg/infrastructure/services/dashboards.go:160]: Yes, but we can leave them. Fixing it

**Comment by r00ta** [comment]
See https://github.com/kiegroup/kogito-cloud-operator/pull/774 for 1.3.x cherry pick of the last change

**Comment by r00ta** [comment]
Hi @spolti , I've changed the regex, if this looks good to you can we move forward with this?

**Comment by ricardozanini** [comment]
@r00ta, can you please check your rebase? I think you brought too many commits.

**Comment by spolti** [comment]
@r00ta sure, but first, as Zanini mentioned, do you mind to check why your latest commit brought all these unwanted changes?

**Comment by r00ta** [comment]
@spolti @ricardozanini , I don't remember I did that (also because I see the `verified` flag on those commits), maybe @sutaakar ?

**Comment by ricardozanini** [comment]
@spolti can you please take a look? I'll merge this one as soon as you finish your review.

**Review by spolti — APPROVED** [review]

**Pull request merged**
